src/Microsoft.AspNetCore.Blazor.Razor.Extensions/BlazorDesignTimeNodeWriter.cs

```diff
@@ -564,37 +657,69 @@ public override void WriteTemplate(CodeRenderingContext context, TemplateInterme
             _scopeStack.CloseScope(context);
         }
 
-        public override void WriteReferenceCapture(CodeRenderingContext context, RefExtensionNode refNode)
+        public override void WriteReferenceCapture(CodeRenderingContext context, RefExtensionNode node)
         {
             if (context == null)
             {
                 throw new ArgumentNullException(nameof(context));
             }
 
-            if (refNode == null)
+            if (node == null)
             {
-                throw new ArgumentNullException(nameof(refNode));
+                throw new ArgumentNullException(nameof(node));
             }
 
-            // The runtime node writer moves the call elsewhere. At design time we
-            // just want sufficiently similar code that any unknown-identifier or type
-            // errors will be equivalent
-            var captureTypeName = refNode.IsComponentCapture
-                ? refNode.ComponentCaptureTypeName
-                : BlazorApi.ElementRef.FullTypeName;
-            WriteCSharpCode(context, new CSharpCodeIntermediateNode
+            WriteReferenceCaptureInnards(context, node, shouldTypeCheck: true);
+            context.CodeWriter.WriteLine(";");
+        }
+
+        protected override void WriteReferenceCaptureInnards(CodeRenderingContext context, RefExtensionNode node, bool shouldTypeCheck)
+        {
+            // We specialize this code based on whether or not we can type check. When we're calling into
+            // a type-inferenced component, we can't do the type check. See the comments in WriteTypeInferenceMethod.
+            if (shouldTypeCheck)
+            {
+                // The runtime node writer moves the call elsewhere. At design time we
```

Comment:
This is another case where what we're doing at design time is specialized and can't work with generics. I preserved the exising design time behaviour when you don't need inference.